src/content/content-safari.js

```diff
@@ -208,11 +214,102 @@ const handleLogoutRequest = async function handleLogoutRequest (message) {
   uiController.api.auth.logout()
 }
 
-const sendMessageToBackground = function sendStateToBackground (messageName) {
+const sendMessageToBackground = function sendMessageToBackground (messageName) {
   safari.extension.dispatchMessage(messageName, new StateMessage(state))
 }
 
 document.addEventListener('DOMContentLoaded', (event) => {
+  // Handle the Auth0 logout page
+  if (window.document.URL.includes(env.AUTH0_LOGOUT_URL)) {
+    // Empty the logout page
+    window.document.body.innerHTML = ''
+    /*
+    This is an Auth0 logout page. We need to navigate away from it.
+    Logout is happening in a frame with a log in button.
+    Because of this we need to set auth to 1 so that the log out button
+    will be rendered in that frame.
+     */
+    window.document.location.replace(`${env.ALPHEIOS_DOMAIN}#auth=1`)
+    return
+  }
+
+  if (window.document.URL.includes(env.ALPHEIOS_DOMAIN)) {
+    const searchRes = window.document.URL.match(/auth=(\d+)/)
+    if (!searchRes) {
+      // Do not inject anything if `auth` is not present
+      return
+    }
+    const authStatus = (searchRes && searchRes.length === 2 && searchRes[1] === '1')
+    let authDiv = document.createElement('div') // eslint-disable-line prefer-const
+
+    authDiv.innerHTML += authStatus
+      ? '<button id="alpheios-logout" class="alpheios-button-primary">Log Out</button>'
```

Comment:
we should retrieve these strings from the messages file

Reply:
Agree with that. We can probably do that as the next iteration (i.e. I can do another PR with those changes).

I was thinking about moving the whole code part that renders the UI in the iframe to the components. Maybe event creating a Vue UI component (effectively just a button) that will be pulled out from components by webextension and then mounted within an iframe. But maybe that'll be to much. What do you think?

Reply:
yes, probably a good idea. But let's wait until after we release this.